chain-impl-mockchain/src/vote/payload.rs

```diff
@@ -96,6 +138,9 @@ mod tests {
         fn arbitrary<G: Gen>(g: &mut G) -> Self {
             match PayloadType::arbitrary(g) {
                 PayloadType::Public => Payload::public(Choice::arbitrary(g)),
+                PayloadType::Private => {
+                    unimplemented!("Arbitrary is not implemented for private votes")
+                }
```

Comment:
I don't even think it can be implemented for now because all of them should be signed with the same key or it will be futile since they won't work together.

Reply:
How is it used in tests exactly? If tests work _only_ with vectors of those values, then we could implement `Arbitrary for Vec<Payload>` instead and use the same vote encryption key for all payloads.

Reply:
Thats a good idea. How would we control the size of the vec in that case? Could we implement it for some `[Payload; T]`?
What do you think?

Reply:
That's a bit more complicated. AFAIK Rust will require you to implement it for all lengths you want, like `[Payload; 3]`, `[Payload; 5]`, since const generics are not there yet.

Reply:
You could always write a macro for that though.

Reply:
So the thing is that an arbitrary size of votes may be overkilling IMO

Reply:
Strictly speaking, I don't think think that it is an overkill. It's just the language preventing us from expressing this in some nice way.

Reply:
Yeah, but I mean. I don't feel comfortable not controlling what is the size of the sample.

Reply:
I don't know how is this used, but for a single instance, you could define a constant/static array of encryption keys, and sample from those. Maybe with a non-uniform distribution to make it probable to get many payloads signed with the same one. 
(On a side note, I always find easier to work with `proptest`, as you can just pass parameters).